src/l-tile-layer-wms.js

```diff
@@ -0,0 +1,56 @@
+// @ts-check
+import { tileLayer } from "leaflet";
+import LLayer from "./l-layer.js";
+import { layerConnected } from "./events.js";
+import { htmlAttribute, optional, parse, partial } from "./parse.js";
+
+class LTileLayerWMS extends LLayer {
+  constructor() {
+    super();
+    this.layer = null;
+  }
+
+  connectedCallback() {
+    const urlTemplate = parse(htmlAttribute("url-template"), this);
+
+    const name = this.getAttribute("name");
+    const schema = partial({
+      // Leaflet.tileLayer default options: https://leafletjs.com/reference.html#tilelayer-wms-layers
+      layers: htmlAttribute("layers"),
+      styles: optional(htmlAttribute("styles")),
+      format: optional(htmlAttribute("format")),
+      transparent: optional(htmlAttribute("transparent")),
+      version: optional(htmlAttribute("version")),
+      crs: optional(htmlAttribute("crs")),
+      uppercase: optional(htmlAttribute("uppercase")),
+
+      // Inherited option from Layer: https://leafletjs.com/reference.html#tilelayer-wms-attribution
+      attribution: optional(htmlAttribute("attribution")),
+    });
+
+    const standardOptions = parse(schema, this);
+    const nonStandardOptionsElement = this.getAttribute("options");
+    const nonStandardOptions = () => {
+      try {
+        return JSON.parse(nonStandardOptionsElement);
+      } catch (e) {
+        console.error(
+          "Error whilst parsing JSON for options attribute in l-tile-layer-wms",
+          e,
+        );
+        return null;
```

Comment:
How does this behave with `...` destructuring?

Reply:
great point, addressed in two commits:
- [208d045](https://github.com/andrewgryan/leaflet-html/pull/50/commits/208d0454ed5594597b36048f555cf5321abf65ed)
- [9211860](https://github.com/andrewgryan/leaflet-html/pull/50/commits/92118606bb0ddbb894eb3cf4557db98b88a3216d)